zbus/src/connection/mod.rs

```diff
@@ -1217,6 +1233,56 @@ impl Connection {
             .map_err(Into::into)
     }
 
+    /// Gracefully close the connection, waiting for all other references to be dropped.
+    ///
+    /// This will not disrupt any incoming or outgoing method calls, and will await their
+    /// completion.
+    ///
+    /// # Caveats
+    ///
+    /// * Calling this will not prevent new incoming messages from keeping the collection alive, so
+    ///   you may want to take precautions like deregistering well-known names to reduce the
+    ///   likelihood of this occurring.
+    ///
+    /// * This shutdown will not complete until the underlying connection is fully dropped, so
+    ///   beware of deadlocks if you are holding any other clones of this `Connection`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use std::error::Error;
+    /// use zbus::connection::Builder;
+    /// use zbus::interface;
+    ///
+    /// struct MyInterface;
+    ///
+    /// #[interface(name = "foo.bar.baz")]
+    /// impl MyInterface {
+    ///     async fn do_thing(&self) {}
+    /// }
+    ///
+    /// async fn graceful_example() -> Result<(), Box<dyn Error>> {
```

Comment:
Great job with adding a simple example here. Some notes still:

* Best to [hide non-relevant code](https://github.com/dbus2/zbus/blob/cbeedc62ec9b27867e5fa8ca839a46f4a91b10ab/zbus/src/connection/mod.rs#L842) since this is strictly about the `shutdown` method. 
* Let's use a `main` method so that the code is actually run as a test.

Reply:
I learned something about doc-tests today :)

Done.